### PR TITLE
Updates dependencies

### PR DESCRIPTION
**Pull request by dcalvoalonso**
This PR address the following points:
- Updates dependencies https://github.com/telefonicaid/lwm2m-node-lib/issues/134
- Removes support to Node.js https://github.com/telefonicaid/lwm2m-node-lib/issues/137

### CODE REVIEW

**Comment by fgalan** [comment]
I'd suggest merging PR telefonicaid/iotagent-json#339 first, then review this one (as the resolution of telefonicaid/iotagent-json#339 could influence in this one).

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -7,3 +7,24 @@
 - Update COAP librery to version 0.21.0
 - Using precise dependencies (~=) in packages.json
 - Update MongoDB to 2.2.35
+- Fix: updating dependencies due to known vulnerabilities in the previous ones
+	async: 1.5.2 -> 2.6.1
+    underscore: 1.7.0 -> 1.9.1
+    logops: 0.5.0 -> 1.0.0
+    coap: 0.5.0 -> 1.0.0
+    coap-cli: 0.5.0 -> 1.0.0
+    mongodb: 2.2.35 -> 3.1.8
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: mongodb and mongoose dependencies has changed in this PR. Has the comment at https://github.com/telefonicaid/iotagent-node-lib/pull/684#issuecomment-435838531 been taken into account? Have you tested with a MongoDB replica set as DB to see if it works? (I'm afraid that Travis is not so flexible to do that... but in a local environment it should be possible).

File: CHANGES_NEXT_RELEASE [hunk 2]
```diff
@@ -7,3 +7,24 @@
...
+    mongoose: 4.4.3 -> 5.3.9
+- Remove: unmaintained dependency with mongoose-auto-increment
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: I understand mongo-plugin-autoinc is a kind of replacement of mongoose-auto-increment. As alternative you could replace these 2 lines by:

```
- Replace mongoose-auto-increment dependency (unmantained) with mongoose-plugin-autoinc
```

**Comment by fgalan** [comment]
I understand the PR is still in WIP status (as travis for v4 seems to be failing) but I have provide some early feedback in the form of comments that I hope can be useful.

**Comment by dcalvoalonso** [comment]
Thanks for the early feedback @fgalan! Yes, I am still working on it. I hope to achieve a solution that works for the different versions of Node.js

File: CHANGES_NEXT_RELEASE (hunk 2 above)
Comment (dcalvoalonso) [reply CHANGES_NEXT_RELEASE]: Fixed with 18fc9c6da0ff51552ac0068654c017b953354440

File: CHANGES_NEXT_RELEASE (hunk 1 above)
Comment (dcalvoalonso) [reply CHANGES_NEXT_RELEASE]: Fixed with b3fd0fd744c4c6c0131fb4a33118744aaa978a4d

Comment (dcalvoalonso) [reply CHANGES_NEXT_RELEASE]: I have just tested this with https://github.com/telefonicaid/lightweightm2m-iotagent/pull/141 agains a replica set in my local development environment and it works as you can see in the attached screenshot.

https://github.com/telefonicaid/lwm2m-node-lib/commit/b3fd0fd744c4c6c0131fb4a33118744aaa978a4d just implements the same changes of https://github.com/telefonicaid/iotagent-node-lib/pull/695

lwm2m_replica_mongo

**Comment by dcalvoalonso** [comment]
@fgalan I think that this PR is ready to get your final comments and review. :)

File: CHANGES_NEXT_RELEASE (hunk 1 above)
Comment (fgalan) [reply CHANGES_NEXT_RELEASE]: Looks nice. In addition, I see the changes in lib/services/model/dbConn.js are the same at https://github.com/telefonicaid/iotagent-node-lib/pull/695.

NTC

File: docs/deprecated.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# Deprecated functionality
```
Comment (fgalan) [inline docs/deprecated.md]: Is this document linked from the main README.md? If not, it should

File: docs/deprecated.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+A list of deprecated features and the version in which they were deprecated follows:
+
+* Support to Node.js v4 in lwm2m-node-lib 1.1.0.
+
+## Using old lwm2m-node-lib versions
+
+Although you are encouraged to use always the newest lwm2m-node-lib version, take into account the following
+information in the case you want to use old versions:
+
+* Code corresponding to old releases is
```
Comment (fgalan) [inline docs/deprecated.md]: Better to use examples meaninfull in the case of this repo. Just replace 0.23.0 by 0.5.0 for instance.

File: docs/deprecated.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  (e.g. 0.23.0) has associated the following:
+	* A tag, e.g. `0.23.0`. It points to the base version.
+	* A release branch, `release/0.23.0`. The HEAD of this branch usually matches the aforementioned tag. However, if some
+    hotfixes were developed on the base version, this branch contains such hotfixes.
+
+The following table provides information about the last lwm2m-node-lib version supporting currently removed features:
+
+| **Removed feature**                                                        | **Last lwm2m-node-lib version supporting feature** | **That version release date**   |
+|----------------------------------------------------------------------------|-------------------------------------------|---------------------------------|
+| Support to Node.js v4                                                      | 1.0.0                                     | July 12th, 2018                  |
```
Comment (fgalan) [inline docs/deprecated.md]: Looking to this I have realized that although we tagged 1.0.0, we never defined a release for that at https://github.com/telefonicaid/lwm2m-node-lib/releases :)

I'll fix it

Comment (fgalan) [reply docs/deprecated.md]: Done at https://github.com/telefonicaid/lwm2m-node-lib/releases/tag/1.0.0

NTC

File: docs/deprecated.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+| Support to Node.js v6                    | Not yet defined                 | Not yet defined
```
Comment (fgalan) [inline docs/deprecated.md]: Maybe better to say at second column:

"Not yet defined but it will be done by May 2019"

File: .travis.yml [hunk 7]
```diff
@@ -16,5 +15,5 @@ services:
 install:
  - npm install
 
-before_install:
- - npm install -g grunt-cli
+before_script:
+ - npm run lint
```
Comment (fgalan) [inline .travis.yml]: Great to add lint to the .travis.yml checks.

However, it seems no specific fix in this PR has been done regarding linting. Maybe I have miss it, maybe the lint script is not working properly or maybe the code was already perfect from the point of linting :) What do you think?

Just to be sure...

File: docs/deprecated.md (hunk 4 above)
Comment (dcalvoalonso) [reply docs/deprecated.md]: Fixed with 1bbaed44caf152c9cd022154525162218accdfab

File: docs/deprecated.md (hunk 6 above)
Comment (dcalvoalonso) [reply docs/deprecated.md]: Fixed with 1bbaed44caf152c9cd022154525162218accdfab

File: docs/deprecated.md (hunk 3 above)
Comment (dcalvoalonso) [reply docs/deprecated.md]: Fixed with 1bbaed44caf152c9cd022154525162218accdfab and 6f89145c46d949c6ae37b4ac499313c28bf9449e

File: .travis.yml (hunk 7 above)
Comment (fgalan) [reply .travis.yml]: Great. NTC

File: docs/deprecated.md (hunk 5 above)
Comment (fgalan) [reply docs/deprecated.md]: (Strictly speaking, release has been created on Nov 12th. July 12th is the date of the tag. However, it is just a little detail)

**Pull request merged**